crates/wasi/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
+name = "wasmi_wasi"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+
+wasi-common = "2.0.1"
+wasi-cap-std-sync = "2.0.1"
+wiggle = "2.0.0"
```

Comment:
To my limited understanding the `wiggle` crate is specific to the unstable Wasm component model. So do we actually need it if we have no support for the unstable Wasm component model in the entire rest of the codebase? (https://crates.io/crates/wiggle)

Reply:
So, I'll try to make the case for it.
1. `wasi-common` is implemented with it's macros. The readme has this to say:

> The [wasi-common crate](https://github.com/bytecodealliance/wasmtime/tree/main/crates/wasi-common) is implemented using Wiggle and the [wasmtime-wasi crate](https://github.com/bytecodealliance/wasmtime/tree/main/crates/wasi) integrates wasi-common with the Wasmtime engine

3. I suppose I could find workarounds. But they'll be workarounds. Also, `preview_2` is in the works, and I think it's coming with it's own IDL that isn't `witx`. But as long as `preview_1` is defined in `witx`, the rust code generated by `wiggle`'s `from_witx` macro is all I have to work with.  And that interface exposes as part of it's definition a few wiggle types I have to use/convert from. I'm going on a limb here to say that to use `wasi_common`'s snapshots meaningfully, you'd have to use wiggle too.

Look for instance at one of the functions in `wasi preview_1` as defined by `wasi_common`:
```rust
pub fn path_create_directory<'a>(
    ctx: &'a mut impl [WasiSnapshotPreview1](https://docs.rs/wasi-common/2.0.2/wasi_common/snapshots/preview_1/wasi_snapshot_preview1/trait.WasiSnapshotPreview1.html) + [UserErrorConversion](https://docs.rs/wasi-common/2.0.2/wasi_common/snapshots/preview_1/types/trait.UserErrorConversion.html),
    memory: &'a dyn [GuestMemory](https://docs.rs/wiggle/2.0.2/x86_64-unknown-linux-gnu/wiggle/trait.GuestMemory.html),
    arg0: [i32](https://doc.rust-lang.org/nightly/std/primitive.i32.html),
    arg1: [i32](https://doc.rust-lang.org/nightly/std/primitive.i32.html),
    arg2: [i32](https://doc.rust-lang.org/nightly/std/primitive.i32.html)
) -> impl [Future](https://doc.rust-lang.org/nightly/core/future/future/trait.Future.html)<Output = [Result](https://doc.rust-lang.org/nightly/core/result/enum.Result.html)<[i32](https://doc.rust-lang.org/nightly/std/primitive.i32.html), [Trap](https://docs.rs/wiggle/2.0.2/x86_64-unknown-linux-gnu/wiggle/enum.Trap.html)>> + 'a
```

But maybe I can do without it. I'll work on that and let you know. But my little understanding is if we're cursed with wasi_common, we're cursed with wiggle. I'll look more deeply into it. Prolly missing something